demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -976,11 +996,26 @@ static int establishMqttSession( MQTTContext_t * pMqttContext,
      * reestablish a session which was already present. */
     connectInfo.cleanSession = createCleanSession;
 
-    /* The client identifier is used to uniquely identify this MQTT client to
+    /* Generate a random number to prefix to the client ID string.
+     * The randomization of the client ID is used to avoid
+     * client identifier collisions when connecting to the MQTT broker. */
+    randomNumForClientId = ( rand() % ( MAX_RAND_NUMBER_FOR_CLIENT_ID + 1u ) );
+
+    /* Create the client ID string with the random number.
+     * The client identifier is used to uniquely identify this MQTT client to
      * the MQTT broker. In a production device the identifier can be something
      * unique, such as a device serial number. */
-    connectInfo.pClientIdentifier = CLIENT_IDENTIFIER;
-    connectInfo.clientIdentifierLength = CLIENT_IDENTIFIER_LENGTH;
+    connectInfo.clientIdentifierLength =
+        snprintf( clientIdBuffer, sizeof( clientIdBuffer ),
+                  "%d%s",
+                  randomNumForClientId,
+                  CLIENT_IDENTIFIER );
+    connectInfo.pClientIdentifier = clientIdBuffer;
+
+    LogDebug( ( "Created randomized client ID for MQTT connection: "
+                "ClientID={%.*s}",
```

Comment:
This format appears to be dependent on the underlying log buffer size, the size of the input, and the previous code preventing malformed data being passed. Is the underlying log implementation defensive against buffer size issues?